packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts

```diff
@@ -94,6 +94,11 @@ export interface UseFieldInternalProps<
    * @default "dense"
    */
   formatDensity?: 'dense' | 'spacious';
+  /**
+   * If `true`, doesn't allow the selection of ranges containing disabled dates.
+   * @default false
+   */
+  disableNonContiguousRanges?: boolean;
```

Comment:
@LukasTy @arthurbalduini do you think `disable` is the right verb here?
Wouldn't `refuse` or `invalidate` be more correct? 
We don't disable them, we ask the validation to refuse them.

Reply:
Agree it can be weird from the Fields perspective

On the other hand, for Calendar component it is indeed a disablement, no ?

Also on this topic: I'm not very satisfied with the double negative (disable + non). WDYT about going with an affirmative prop name with default value set to `false`? Something similar to what is done by [react spectrum](https://react-spectrum.adobe.com/react-spectrum/RangeCalendar.html#non-contiguous-ranges)

Reply:
> WDYT about going with an affirmative prop name with default value set to false? 

Changing the default behavior in a minor is too drastic I think
If we want to disallow non contiguous date range by default, then I would set the default value to `true` until the next major.

___

> On the other hand, for Calendar component it is indeed a disablement, no ?

On the view yes
We have `shouldDisableDate` so I guess `disable` is not totally wrong... even if I prefer React Spectrum naming (allow / disallow, and `isDateUnavailable` instead of `shouldDisableDate` but it's another topic)

Reply:
> Changing the default behavior in a minor is too drastic I think
> If we want to disallow non contiguous date range by default, then I would set the default value to true until the next major.
I believe it's safer to keep allowing non contiguous by default, avoiding unexpected behavior changes for current users

Another naming matter that just ocurred me is to explicit that the prop regards to "Date": something like `disableNonContiguousDateRange`. If tomorrow we want to implement a similar feature for time ranges, the current name of the prop is generic and we could must avoid re-naming it again. I will raise this discussion on the channel to collect insights from the team.